db/statistics.go

```diff
@@ -26,266 +29,435 @@ func WriteValidatorStatisticsForDay(day uint64) error {
-			valueArgs = append(valueArgs, stat.OrphanedAttestations)
-		}
-		stmt := fmt.Sprintf(`
-			insert into validator_stats (validatorindex, day, missed_attestations, orphaned_attestations) VALUES
-			%s
-			on conflict (validatorindex, day) do update set missed_attestations = excluded.missed_attestations, orphaned_attestations = excluded.orphaned_attestations;`,
-			strings.Join(valueStrings, ","))
-		_, err := tx.Exec(stmt, valueArgs...)
-		if err != nil {
-			return err
-		}
+	if err := CheckIfDayIsFinalized(day); err != nil {
+		return err
+	}
+
+	start := time.Now()
+	logger.Infof("validating if required data has been exported for total performance")
+	type Exported struct {
+		LastClRewards    bool `db:"last_cl_rewards_exported"`
+		LastElRewards    bool `db:"last_el_rewards_exported"`
+		CurrentCLRewards bool `db:"cur_cl_rewards_exported"`
+		CurrentElRewards bool `db:"cur_el_rewards_exported"`
+	}
+	exported := Exported{}
+	err := ReaderDb.Get(&exported, `
+		SELECT 
+			last.cl_rewards_exported as last_cl_rewards_exported, 
+			last.el_rewards_exported as last_el_rewards_exported, 
+			cur.cl_rewards_exported as cur_cl_rewards_exported, 
+			cur.el_rewards_exported as cur_el_rewards_exported
+		FROM validator_stats_status cur
+		INNER JOIN validator_stats_status last 
+				ON last.day = GREATEST(cur.day - 1, 0)
+		WHERE cur.day = $1;
+	`, day)
 
-		logger.Infof("saving failed attestations batch %v completed", b)
+	if err != nil {
+		return fmt.Errorf("error retrieving required data: %v", err)
+	} else if !exported.CurrentCLRewards || !exported.CurrentElRewards || !exported.LastClRewards || !exported.LastElRewards {
+		return fmt.Errorf("missing required export: cur cl rewards: %v, cur el rewards: %v, last cl rewards: %v, last el rewards: %v", !exported.CurrentCLRewards, !exported.CurrentElRewards, !exported.LastClRewards, !exported.LastElRewards)
 	}
+	logger.Infof("validating completed, took %v", time.Since(start))
 
+	tx, err := WriterDb.Beginx()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	start = time.Now()
+
+	logger.Infof("exporting total income stats")
+	_, err = tx.Exec(`
+	INSERT INTO validator_stats (validatorindex, day, cl_rewards_gwei_total, cl_proposer_rewards_gwei_total, el_rewards_wei_total, mev_rewards_wei_total) (
+		SELECT 
+			vs1.validatorindex, 
+			vs1.day, 
+			COALESCE(vs1.cl_rewards_gwei, 0) + COALESCE(vs2.cl_rewards_gwei_total, 0) AS cl_rewards_gwei_total_new, 
+			COALESCE(vs1.cl_proposer_rewards_gwei, 0) + COALESCE(vs2.cl_proposer_rewards_gwei_total, 0) AS cl_proposer_rewards_gwei_total_new, 
+			COALESCE(vs1.el_rewards_wei, 0) + COALESCE(vs2.el_rewards_wei_total, 0) AS el_rewards_wei_total_new, 
+			COALESCE(vs1.mev_rewards_wei, 0) + COALESCE(vs2.mev_rewards_wei_total, 0) AS mev_rewards_wei_total_new 
+		FROM validator_stats vs1 LEFT JOIN validator_stats vs2 ON vs2.day = vs1.day - 1 AND vs2.validatorindex = vs1.validatorindex WHERE vs1.day = $1
```

Comment:
**Minor:** `vs1` and `vs2` aren't the best names.
Here you use `cur` and `last` (better) for example:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/401df27e99a88a6138ed1b8b8ee3911e93a67fba/db/statistics.go#L579-L583